tests/test_uploads.py

```diff
@@ -0,0 +1,364 @@
+# -*- coding: utf-8 -*-
+import math
+import sys
+# import mock
+import http.client
+from hashlib import md5
+from unittest import skip, TestCase
+from unittest.mock import patch, mock_open, Mock
+from http.client import RemoteDisconnected
+
+from jwplatform import __version__
+from jwplatform.client import JWPlatformClient, JWPLATFORM_API_HOST
+import logging
+
+# from .mock import JWPlatformMock
+from jwplatform.upload import UploadType, MaxRetriesExceededError, S3UploadError, PartUploadError
+from tests.mock import JWPlatformMock
+from tests.s3mock import S3Mock
+
+
+def _get_parts_responses(part_count):
+    parts = [{"upload_link": "http://s3server/upload-link",
+              'id': part_id + 1} for part_id in range(part_count)]
+    result = {'page': 1, 'page_length': 10, 'total': 10, 'parts': parts}
+    return result
+
+
+class TestUploads(TestCase):
+    file_content_mock_data_simple = b'some bytes'
+    file_content_mock_data_large = b'some bytes' * 5 * 1024 * 1024
+
+    @patch("os.stat")
+    @patch("builtins.open")
+    def test_upload_method_is_direct_when_file_size_is_small(self, mock_file, os_stat):
+        os_stat.return_value.st_size = 4 * 1024 * 1024
+        site_id = 'siteDEid'
+        client = JWPlatformClient()
+        media_client_instance = client.Media
+        file_absolute_path = "mock_file_path"
+        logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
```

Comment:
Could also probably do logging stuff as a part of test setup.

Reply:
Done